yaksh/models.py

```diff
@@ -383,6 +383,20 @@ def get_failed_students(self, course):
                 course=course, passed=False
             ).values_list("user", flat=True).distinct().count()
 
+    def get_answerpaper_status(self, user, course):
+        try:
+            qp = self.questionpaper_set.get().id
+        except QuestionPaper.DoesNotExist:
+            qp = None
+        ans_ppr = AnswerPaper.objects.filter(
+            user=user, course=course, question_paper=qp
+        ).order_by("-attempt_number")
+        if ans_ppr.exists():
+            status = ans_ppr.first().status
+        else:
```

Comment:
Can't we just let status be returned as `None`? we can check that instead of checking `if status == "not attempted"?`

Reply:
This `not attempted` is used to show on the interface. So I feel it is okay to send from models. If I make it None then I need to add a condition in the template.

Reply:
As explained previously, `not attempted` is used at various places to show the status of a quiz or lesson or module. So instead of adding conditions in multiple templates, I directly pass the status from model methods.